### PR TITLE
Implemented showing and hiding usernames in the submissions page

### PR DESCRIPTION
**Pull request by grobalex**
Related to Issue #137

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -109,7 +109,10 @@ def show_user(user)
   end
   def maybe_link_user(show, user, extra_class="")
     if show == true || (show.is_a?(Array) && show.member?(user.id))
-      link_to(user.display_name, user_path(user), class: "user-link #{extra_class}".strip, data: user_link_data(user))
+      content_tag(:span,[
+          link_to(user.display_name, user_path(user), class: "user-link #{extra_class}".strip, data: user_link_data(user)),
+          content_tag(:span, "(#{user.username})", class: "username")
+        ].flatten.join(" ").html_safe)
     else
       content_tag :span, user.name, class: "user-link #{extra_class}".strip, data: user_link_data(user)
```
Comment (blerner) [inline app/helpers/application_helper.rb]: I think we need to do the same thing to this span as we did to the link above, and wrap it in a further span containing the "(username)" text...

File: app/views/assignments/_admin_subs.html.erb [hunk 2]
```diff
@@ -274,6 +278,10 @@
         widgets: ['uitheme']
       });
       $(".no-resort").on("mousedown", function(e) { e.stopPropagation(); });
+      $("#show_username").click(function(){
```
Comment (blerner) [inline app/views/assignments/_admin_subs.html.erb]: Let's move this code into a function in app/assets/javascripts/application.js,
```js
function enableShowUsernames(elts) {
  $(elts).click(....);
}
```
and then it's callable as `enableShowUsername("#show_username")` from this file.  That way, we can enable this in more places more easily.

File: app/views/assignments/_admin_subs.html.erb [hunk 3]
```diff
@@ -132,14 +132,18 @@
...
-        <% else %>
-        <th>Submitter</th>
-        <% end %>
+        <th>
+          <% if @gradesheet.assignment.team_subs? %>
+          Team (submitter in bold)
+          <% else %> 
+          Submitter
+          <% end %>
+          <button class="btn btn-default no-resort" id="show_username" title="Show/hide usernames" data-target="submissions">
```
Comment (blerner) [inline app/views/assignments/_admin_subs.html.erb]: We probably want to turn this into a helper in app/helpers/application_helper.rb, 
```ruby
def toggle_usernames_button(id, target, class: nil)
  button_tag(type: nil, class: (class || "btn btn-default no-resort"), title: "Show/hide usernames", id: id, data: {target: target}) do
    content_tag :span, class: "glyphicon glyphicon-user", ""
  end
end
```
that can then be called as `<%= toggle_usernames_button("show_username", "submissions") %>`

File: app/views/assignments/_admin_subs.html.erb (hunk 2 above)
Comment (grobalex) [reply app/views/assignments/_admin_subs.html.erb]: I have moved it into app/assets/javascripts/application.js 
Is there a specific way Rails handles JS files, as it complains that it cannot find enableShowUsername in  app/views/assignments/_admin_subs.html.erb

**Pull request merged**